src/components/message-input/index.tsx

```diff
@@ -341,6 +341,10 @@ export class MessageInput extends React.Component<Properties, State> {
     const hasInputValue = this.state.value?.length > 0;
     const reply = this.props.reply;
 
+    if (this.props.reply) {
```

Comment:
For some reason that I haven't quite been able to pin point, a timeout is required here to ensure the input is focussed when replying to a message. Even if we update componentDidUpdate like so 

```
  componentDidUpdate(prevProps: Properties) {
    if (this.props.reply && !prevProps.reply) {
      this.focusInput();
    }

    if (this.props.onMessageInputRendered) {
      this.props.onMessageInputRendered(this.textareaRef);
    }
    if (this.state.isSendTooltipOpen && this.isSendingEnabled) {
      this.setState({ isSendTooltipOpen: false });
    }
  }
  ```

and log 

```
 focusInput() {
    console.log('Textarea ref:', this.textareaRef.current);
    if (this.textareaRef && this.textareaRef.current) {
      this.textareaRef.current.focus();
    }
  }
  ```
 the textareaRef is logged and we can see the textarea is the current focus, but the cursor is not active. The timeout seems to fix this.

Reply:
Hmm... I don't think it makes sense to do this in the render function. This function can get called thousands of times at any point the app decides something needs to be rerendered. This will cause the input to constantly steal focus away even if you've clicked around to somewhere else.

Reply:
Spent some time digging into this more today. I believe this is *another* radix ref issue. When selecting the `reply` option in the dropdown menu, the active element (even if we set the focus to the textarea) is the dropdown menu option, or it can be the dropdown menu trigger. Seems like the timeout I have added, (replaced from render function now to `focusInput` function) is a fix for the actions that are queued in the event loop. Hacky, but seems a good workaround in this instance.

Reply:
Here I am console logging  `console.log(document.activeElement, this.textareaRef.current);`. You can see the `this.textareaRef.current` is the expected textarea, but the active element remains the menu option. Selecting an emoji also uses the `focusInput` function, and as you can see in the demo here the activeElement is the textarea. A `e.stopPropagation` in `DropdownMenu` may help with this, but from digging around I believe this is due to refs controlled on the Radix side of things.

https://github.com/zer0-os/zOS/assets/39112648/07ea2a4a-7ceb-414d-a884-429e1a599d4b